lib/fog/compute/ecloud.rb

```diff
@@ -356,6 +356,9 @@ def set_extra_headers_for(params)
             "x-tmrk-version" => @version,
             "Date"           => Time.now.utc.strftime("%a, %d %b %Y %H:%M:%S GMT"),
           }.merge(params[:headers] || {})
+          if params[:body].size > 0 || (params[:method] == 'POST' || params[:method] == 'PUT')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.